pkg/util/iptables/iptables.go

```diff
@@ -328,10 +328,11 @@ func (runner *runner) checkRuleWithoutCheck(table Table, chain Chain, args ...st
 
 	// Sadly, iptables has inconsistent quoting rules for comments.
 	// Just unquote any arg that is wrapped in quotes.
-	argsCopy := make([]string, len(args))
-	copy(argsCopy, args)
-	for i := range argsCopy {
-		unquote(&argsCopy[i])
+	var argsCopy []string
+	for i := range args {
+		tmpField := args[i]
+		unquote(&tmpField)
+		argsCopy = append(argsCopy, strings.Fields(tmpField)...)
```

Comment:
No problem. Glad I can help.

FYI, I do not think the dupe rules reach line 358.
I think they break out of the loop around line 343 where len(fields) != len(args)+2.

Following is log of kube-proxy --v=5 when adding dupe rules.
```
I0819 16:12:46.112720     534 iptables.go:193] running iptables -N [KUBE-PORTALS-CONTAINER -t nat]
I0819 16:12:46.118800     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.125400     534 iptables.go:193] running iptables -I [PREROUTING -t nat -m comment --comment handle ClusterIPs; NOTE: this must be before the No
dePort rules -j KUBE-PORTALS-CONTAINER]
I0819 16:12:46.128778     534 iptables.go:193] running iptables -N [KUBE-PORTALS-HOST -t nat]
I0819 16:12:46.134010     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.139270     534 iptables.go:193] running iptables -I [OUTPUT -t nat -m comment --comment handle ClusterIPs; NOTE: this must be before the NodePo
rt rules -j KUBE-PORTALS-HOST]
I0819 16:12:46.142075     534 iptables.go:193] running iptables -N [KUBE-NODEPORT-CONTAINER -t nat]
I0819 16:12:46.147363     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.152393     534 iptables.go:193] running iptables -A [PREROUTING -t nat -m addrtype --dst-type LOCAL -m comment --comment handle service NodePorts; NOTE: this must be the last rule in the chain -j KUBE-NODEPORT-CONTAINER]
I0819 16:12:46.159039     534 iptables.go:193] running iptables -N [KUBE-NODEPORT-HOST -t nat]
I0819 16:12:46.164323     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.169335     534 iptables.go:193] running iptables -A [OUTPUT -t nat -m addrtype --dst-type LOCAL -m comment --comment handle service NodePorts; NOTE: this must be the last rule in the chain -j KUBE-NODEPORT-HOST]
I0819 16:12:46.176321     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.181607     534 iptables.go:251] DBG: fields is not a superset of args: fields=[-A KUBE-PORTALS-CONTAINER -d 192.168.20.1/32 -p tcp -m comment --comment default/kubernetes: -m tcp --dport 443 -j REDIRECT --to-ports 33995]  args=[-m comment --comment kube-system/kube-ui: -p tcp -m tcp --dport 80 -d 192.168.20.135/32 -j REDIRECT --to-ports 39301]
I0819 16:12:46.181698     534 iptables.go:251] DBG: fields is not a superset of args: fields=[-A KUBE-PORTALS-CONTAINER -d 192.168.20.160/32 -p tcp -m comment --comment default/my-nginx: -m tcp --dport 80 -j REDIRECT --to-ports 52592]  args=[-m comment --comment kube-system/kube-ui: -p tcp -m tcp --dport 80 -d 192.168.20.135/32 -j REDIRECT --to-ports 39301]
I0819 16:12:46.184776     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.190756     534 iptables.go:251] DBG: fields is not a superset of args: fields=[-A KUBE-PORTALS-HOST -d 192.168.20.1/32 -p tcp -m comment --comment default/kubernetes: -m tcp --dport 443 -j DNAT --to-destination 10.101.61.147:33995]  args=[-m comment --comment kube-system/kube-ui: -p tcp -m tcp --dport 80 -d 192.168.20.135/32 -j DNAT --to-destination 10.101.61.147:39301]
I0819 16:12:46.190885     534 iptables.go:251] DBG: fields is not a superset of args: fields=[-A KUBE-PORTALS-HOST -d 192.168.20.160/32 -p tcp -m comment --comment default/my-nginx: -m tcp --dport 80 -j DNAT --to-destination 10.101.61.147:52592]  args=[-m comment --comment kube-system/kube-ui: -p tcp -m tcp --dport 80 -d 192.168.20.135/32 -j DNAT --to-destination 10.101.61.147:39301]
I0819 16:12:46.194931     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.203319     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.211753     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.217282     534 iptables.go:251] DBG: fields is not a superset of args: fields=[-A KUBE-PORTALS-CONTAINER -d 192.168.20.1/32 -p tcp -m comment --comment default/kubernetes: -m tcp --dport 443 -j REDIRECT --to-ports 33995]  args=[-m comment --comment default/my-nginx: -p tcp -m tcp --dport 80 -d 192.168.20.160/32 -j REDIRECT --to-ports 52592]
I0819 16:12:46.220289     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.225824     534 iptables.go:251] DBG: fields is not a superset of args: fields=[-A KUBE-PORTALS-HOST -d 192.168.20.1/32 -p tcp -m comment --comment default/kubernetes: -m tcp --dport 443 -j DNAT --to-destination 10.101.61.147:33995]  args=[-m comment --comment default/my-nginx: -p tcp -m tcp --dport 80 -d 192.168.20.160/32 -j DNAT --to-destination 10.101.61.147:52592]
I0819 16:12:46.228792     534 iptables.go:217] running iptables-save -t nat
I0819 16:12:46.237128     534 iptables.go:217] running iptables-save -t nat
```